js/builder_graph.js

```diff
@@ -445,20 +445,36 @@ class PowderInputNode extends InputNode {
 
     compute_func(input_map) {
         // TODO: haha improve efficiency to O(n) dumb
-        // also, error handling is missing
         let input = this.input_field.value.trim();
         let powdering = [];
         let errorederrors = [];
         while (input) {
             let first = input.slice(0, 2);
             let powder = powderIDs.get(first);
             if (powder === undefined) {
-                return null;
+                if (first.length > 0) {
+                    errorederrors.push(first);
+                } else {
+                    break;
+                }
             } else {
                 powdering.push(powder);
             }
             input = input.slice(2);
         }
+
+        if (this.input_field.getAttribute("placeholder") != null) {
+            let placeholder = this.input_field.getAttribute("placeholder");
+            if (placeholder.includes(" slots") && parseInt(placeholder[0]) < powdering.length)
```

Comment:
This is fragile and breaks down for items with more than ten slots (e.g. Singularity).

Reply:
I have made it less fragile by using a string split instead, but do you have any other suggestions?